python/essex-config/README.md

```diff
@@ -0,0 +1,106 @@
+# essex-configuration
+
+essex-configuration is a Python library for creating configuration objects that read configuration from different sources. This library supports reading from files, environment variables, and Azure Key Vault.
+
+## Installation
+
+To install the essex-configuration library, use pip:
+
+`pip install essex-configuration`
+
+## Usage
+
+Here’s an example of how to use the essex-configuration library to create a configuration object for connecting to a customer database:
+
+```python
+from typing import Annotated, ClassVar
+from essex_config import Config, ConfigurationField, FieldType
+from essex_config.sources import Source, EnvSource, FileSource, KeyvaultSource
+
+
+class CustomerDatabase(Config):
+    """Class that holds the configuration to connect to the Customer Database."""
+
+    sources: ClassVar[list[Source]] = [
+        FileSource(file_path="pyproject.toml"),
```

Comment:
Should the sources be extrinsic to the model? I feel like this may couple us to closely to specific config files.

Reply:
I think I don't get your point, could you elaborate? 

Here you could define any file you wish, or even load it from an env variable, for example if you do:

```python
class YourConfig(Config):
    sources: ClassVar[list[Source]] = [FileSource(file_path="some_env_var", use_env_var=True)]
    ...
```

It will load the config from the file you defined in some_env_var in your environment variables.

This is a way to define several sources for your configuration. I like it that defining it here makes it explicit that we are loading the values for that configuration from a specific file.